src/graphql.rs

```diff
@@ -1206,6 +1266,84 @@ impl ___Type for QueryType {
     }
 }
 
+fn function_fields(schema: &Arc<__Schema>, volatilities: &[FunctionVolatility]) -> Vec<__Field> {
+    let sql_types = &schema.context.types;
+    let function_name_to_count = Function::function_names_to_count(&schema.context.functions);
+    schema
+        .context
+        .functions
+        .iter()
+        .filter(|func| func.is_supported(&schema.context, &function_name_to_count))
+        .filter(|func| volatilities.contains(&func.volatility))
```

Comment:
I think we need another filter here for if the role has `executable` permission for each function. Could you please add that with an associated test of the behavior?

Reply:
Thanks @olirice for adding this. Resolving.